src/ucp/core/ucp_listener.c

```diff
@@ -42,77 +60,72 @@ void ucp_listener_schedule_accept_cb(ucp_ep_h ep)
 
 static unsigned ucp_listener_conn_request_progress(void *arg)
 {
-    ucp_listener_accept_t            *accept      = arg;
-    const ucp_wireup_sockaddr_priv_t *client_data = accept->wireup_data;
-    ucp_listener_h                   listener     = accept->listener;
-    ucp_unpacked_address_t           client_address;
-    ucp_ep_params_t                  params;
+    ucp_conn_request_h               conn_request = arg;
+    ucp_listener_h                   listener     = conn_request->listener;
+    const ucp_wireup_client_data_t   *client_data = &conn_request->client_data;
+    ucp_worker_h                     worker;
     ucp_ep_h                         ep;
     ucs_status_t                     status;
 
     ucs_trace_func("listener=%p", listener);
 
-    status = ucp_address_unpack(client_data + 1, &client_address);
+    if (listener->conn_cb) {
```

Comment:
is it safe to access listener->conn_cb without lock?

Reply:
should be:
 - it's initialized once on listener creation
 - `ucp_listener_conn_request_progress` is called from main thread and listener must exist and be initialized